src/OmniSharp.Abstractions/Models/v1/Completion/CompletionItem.cs

```diff
@@ -87,7 +87,12 @@ public class CompletionItem
         /// <summary>
         /// Index in the completions list that this completion occurred.
         /// </summary>
-        public int Data { get; set; }
+        public (long CacheId, int Index) Data { get; set; }
```

Comment:
does this work well serialized over the wire? I could imagine some client implementations might have troubles

Reply:
I would expect it should be just fine, it should be serialized as `{ 'Item1': 1, 'Item2': 2 }`.

Reply:
I see. one other thing that jumped to my mind is that it is technically a breaking change on the contract.

It would be better to pass the CacheId as a separate property, this way it would be an additive change and it wouldn't affect the existing completion service implementations

Reply:
I don't know if I agree. This data is only used by the server, not by clients: the vscode implementation just treats this as an `any` type: https://github.com/OmniSharp/omnisharp-vscode/blob/master/src/omnisharp/protocol.ts#L541. Ideally, it would be a totally opaque blob, but that isn't how json works.

Reply:
yes I understand that in the C# extension it is treated this way (and that it was intended to flow back to the server as sort of a "state"), but it is part of a contract after all. we don't follow all the implementations of Omnisharp (we are not even aware of most!) but we try not to have breaking contract changes

Reply:
actually since this is new anyway I am pretty sure *you* are the only user of it so I guess it doesn't matter 😀